internal/terminal/table_test.go

```diff
@@ -0,0 +1,264 @@
+					"nil":    nil,
+				},
+			},
+			expectedHeader: []string{"test"},
+			expectedData: []map[string]string{
+				{
+					"test": "should show up",
+				},
+			},
+			expectedWidths: map[string]int{
+				"test": 14,
+			},
+		},
+	} {
+		t.Run(tc.description, func(*testing.T) {
+			table := newTable(tc.header, tc.data)
+			assert.Equal(t, tc.expectedWidths, table.columnWidths)
+			assert.Equal(t, tc.expectedData, table.data)
+			assert.Equal(t, tc.expectedHeader, table.headers)
+		})
+	}
+}
+
+func TestTableMessage(t *testing.T) {
+	for _, tc := range []struct {
+		description     string
+		header          []string
+		data            []map[string]interface{}
+		expectedMessage string
+		expectedError   string
+	}{
+		{
+			description: "Should return an empty string and an error if the table has no headers",
+			data: []map[string]interface{}{
+				{"test": 123, "this": "456", "data": []string{"7", "8 9", "10!"}},
+			},
+			expectedError: "cannot create a table without headers",
+		},
+		{
+			description: "Should print only a header for a table with no data",
+			header:      []string{"header", "only", "no", "data"},
+			expectedMessage: "\n" + fmt.Sprintf("%s  %s  %s  %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("header"),
+					color.New(color.Bold).SprintFunc()("only"),
+					color.New(color.Bold).SprintFunc()("no"),
+					color.New(color.Bold).SprintFunc()("data"),
+				}...,
+			) + `
+------  ----  --  ----
+`,
+		},
+		{
+			description: "Should return a string representing the table",
+			header:      []string{"one", "two", "three"},
+			data: []map[string]interface{}{
+				{"one": 1, "two": 2.222, "three": "three"},
+				{"one": "1", "two": 2, "three": 3},
+				{"one": "1.0", "two": 2, "three": 3},
+			},
+
+			expectedMessage: "\n" + fmt.Sprintf("%s  %s    %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("one"),
+					color.New(color.Bold).SprintFunc()("two"),
+					color.New(color.Bold).SprintFunc()("three")}...,
+			) + `
+---  -----  -----
+1    2.222  three
+1    2      3    
+1.0  2      3    `,
+		},
+		{
+			description: "Should return rows with values specified in the header, not insert new columns, and not print empty rows",
+			header:      []string{"three", "one", "two"},
+			data: []map[string]interface{}{
+				{},
+				{"one": 1, "two": 2, "three": -3.123},
+				{},
+				{"one": 1, "two": 2, "test": "no"},
+				{},
+				{"one": 1, "should not show up": 2},
+				{},
+				{"two": []int{2, 3}},
+			},
+
+			expectedMessage: "\n" + fmt.Sprintf("%s   %s  %s  ",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("three"),
+					color.New(color.Bold).SprintFunc()("one"),
+					color.New(color.Bold).SprintFunc()("two")}...,
+			) + `
+------  ---  -----
+-3.123  1    2    
+        1    2    
+        1         
+             [2 3]`,
+		},
+		{
+			description: "Should return a table where the longest length is determined by the longest value in the column as a whole",
```

Comment:
this test case could be accomplished with an assert on `expectedColumnWidths` in `TestNewTable` above (and will prove to be a little easier to read without all the string formatting required in the message assertion here)

Reply:
For these tests, I left it mostly to check for the whitespace after the values as well as to see what the result of arrays, maps, etc. look like in this table printing format